### PR TITLE
Simplify and update documentation

### PR DESCRIPTION
**Pull request by CarloLucibello**
Simplify the readme, add missing doc pages, generally reduce redundancies

### CODE REVIEW

File: .github/workflows/Documenter.yml [hunk 1]
```diff
@@ -14,9 +14,8 @@ jobs:
   build:
     runs-on: ${{ matrix.os }}
     strategy:
-      matrix:
-        julia-version: [1]
-        os: [ubuntu-latest]
+      julia-version: 1.6
+      os: ubuntu-latest
```
Comment (johnnychen94) [inline .github/workflows/Documenter.yml:18]: I feel like `1` is a more compatible choice since it automatically uses the latest 1.x julia release.

**Review by johnnychen94 — COMMENTED** [review]
The idea looks good to me, I notice that a large part of it are whitespace changes, it's okay to do so, but could you revert `src/EMNIST/EMNIST.jl` one since it's 100% unrelated whitespace change.

**Comment by johnnychen94** [comment]
Since it's merged, never mind.

**Comment by CarloLucibello** [comment]
sorry, went too much in a hurry, dealing with this documentation stuff makes me want to be done as soon as possible

File: .github/workflows/Documenter.yml (hunk 1 above)
Comment (CarloLucibello) [reply .github/workflows/Documenter.yml:18]: I'll do it

**Comment by CarloLucibello** [comment]
> The idea looks good to me, I notice that a large part of it are whitespace changes, it's okay to do so, but could you revert src/EMNIST/EMNIST.jl one since it's 100% unrelated whitespace change.

That was reducing the indentation by one level, yeah not related to docs, but there was no code change

**Pull request merged**